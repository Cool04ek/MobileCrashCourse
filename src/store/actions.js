const store = require('./store');

export async function getTopics() {
    store.setters.setLoading(true);
    let response = await fetch('https://www.reddit.com/subreddits/default.json');
    let responseJson = await response.json();
    store.setters.setTopics(responseJson.data.children);
    store.setters.setLoading(false);
}

export async function getPosts(url) {
    store.setters.setLoading(true);
    let response = await fetch(`https://www.reddit.com${url}hot.json`);
    let responseJson = await response.json();
    store.setters.setPosts(responseJson.data.children);
    store.setters.setLoading(false);
}
