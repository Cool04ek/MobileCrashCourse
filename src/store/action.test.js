const store = require('./store');
import * as actions from './actions';

describe('action', () => {

    it('should fetch topics', async () => {
        fetch.mockResponse('{"data":{"children":[{"data":{"id":"1"}}]}}');
        await actions.getTopics();

        expect(store.getters.getTopics()).toMatchSnapshot();
    });
})