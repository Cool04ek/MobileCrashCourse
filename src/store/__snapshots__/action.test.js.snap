// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action should fetch topics 1`] = `
Array [
  Object {
    "data": Object {
      "id": "1",
    },
  },
]
`;
